src/CommunityToolkit.Maui.Core/Essentials/SpeechToText/SpeechToTextImplementation.ios.cs

```diff
@@ -66,12 +69,20 @@ async Task<string> InternalListenAsync(CultureInfo culture, IProgress<string>? r
 					{
 						var s = result.BestTranscription.Segments[i].Substring;
 						currentIndex++;
-						recognitionResult?.Report(s);
+						recognitionProgress?.Report(s);
+						OnRecognitionResultUpdated(s);
 					}
 				}
 			}
 		});
+		return Task.CompletedTask;
+	}
 
+	async Task<string> InternalListenAsync(CultureInfo culture, IProgress<string>? recognitionResult, CancellationToken cancellationToken)
+	{
+		recognitionProgress = recognitionResult;
+		getRecognitionTaskCompletionSource ??= new TaskCompletionSource<string>();
+		await StartListeningAsync(culture, CancellationToken.None);
```

Comment:
Why not use the cancellationToken provided by the user?

Reply:
cancellation token is not used in any of those methods. removed it at all